ChatBot/ChatBot/DataDecoder/Protocols/DataDecoderable.swift

```diff
@@ -0,0 +1,12 @@
+//
+//  DataDecoderProtocol.swift
+//  ChatBot
+//
+//  Created by 김준성 on 1/8/24.
+//
+
+import Foundation
+
+protocol DataDecoderable {
+    func decodeData<T: Decodable>(_ data: Data, to type: T.Type) throws -> T
```

Comment:
저도 이러한 스타일로 네트워크 객체를 구현하고 있습니다 ㅎㅎ 너무 좋네요 
위의 코드에서 하나를 수정하면 좋을거같은데요.

```suggestion
    func decodeData<T: Decodable>(_ data: Data) throws -> T
```

위와 같이 함수를 작성하는 것은 어떤가요? 어차피 return 타입에서 T를 추상화하고 있기 때문에 의도적으로 T.Type을 넣어주지 않아도 사용 할 수 있을거같아요.

Reply:
만약 `NetworkService` 내부가 아닌 제네릭이 없는 다른 곳에서 사용해야할 경우,
```swift
decodeData<Model.UserMessage>(data)
```
이런 식으로 사용해야 합니다. 이는 오히려 `NetworkService`만 사용하다가 `DataDecodable` 프로토콜을 채택한 타입을 사용할 때 혼란을 줄 수 있으므로, 지금의 방식이 더 좋다고 생각합니다!